datamade/la-metro-councilmatic
# Re-implement support for "phrase" matching, handle display of single quotes in search terms

## Overview

I goofed up handling of phrases enclosed in quotes when I added the SES integration. This PR adds back proper handling of exact phrases. In summary, it:

- Filters the default queryset for results containing all search terms (e.g., if you search for "metro red line" and "train", you'll only get results containing both terms)
- If searching attachment text, also return results where the attachment text contains both terms
- Repairs the display of single quotes when rendering a pre-existing search term
- Handles quoted search terms when highlighting topics matching the query

### Checklist


### Notes

I'm going to deploy this to the staging site for further QA.

N.b., this PR does _not_ handle single quote parsing on the back end. That is a deeper issue that I'd like to address separately.

## Testing Instructions

Perform the following searches on the staging site:

* Search `metro red line` and confirm you see results, not necessarily containing the full phrase. Toggle to tag and keyword results and confirm you see the correct results.
* Search `"metro red line"` and confirm you only see results containing the full phrase.
* Search `red line` and confirm you see results. Toggle to the keyword results and confirm you see results tagged `Metro Red Line`.
* Search `it's off limits` and confirm the search term is rendered correctly (i.e., no escaped characters) in the search bar.

Handles #640, handles #663.
